backend/graminsta/core/services.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: UTF-8 -*-
+
+"""
+Services for core module.
+"""
+
+from .serializers import UserSerializer
+from .serializers import UserInfoSerializer
+from .models import UserInfo
+
+
+def create_userinfo(validated_data):
+    """Create or update a UserInfo in databaase.
+
+    Parameters
+    ----------
+    validated_data: json format
+        data containing all the details of User
+
+    Returns
+    -------
+    user_info: json format
+        a successfully created User record
+    """
+    userinfo_serializer = UserInfoSerializer(data=validated_data)
+    userinfo_serializer.is_valid(raise_exception=True)
+    user_data = validated_data.pop('user')
+    user = UserSerializer.create(
+        UserSerializer(), validated_data=user_data)
+    user_info = UserInfo.objects.update_or_create(
```

Comment:
Always create, right? Since the `User` is always a new one.

Reply:
The method is called `create_userinfo` used for `register`. A `update_userinfo` can be implemented for updating user's information,

Reply:
I mean, this API creates a new `User` every time you invoke it. So under what circumstances, an `UserInfo` will be updated during a user registration call? I feel that for each user to be registered, this should create a new `UserInfo`. Though you might get same side effects in this case (`update_or_create` and `create`), but it confuses others on why you are calling `update_or_create` instead of `create`.

Reply:
Fixed